### ISSUE TITLE
Error creating S3 bucket ... the region 'us-east-1' is wrong; expecting 'eu-central-1'

### ISSUE THREAD

**Issue by discentem** [cloudposse/terraform-aws-tfstate-backend#88]
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[![Slack Community](https://slack.cloudposse.com/badge.svg)](https://slack.cloudposse.com)

## Describe the Bug
I've set the AWS provider to use `us-east-1` but I'm getting this error when the module tries to create the s3 bucket: 

```
Error creating S3 bucket: AuthorizationHeaderMalformed: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-central-1'
```

## Expected Behavior
The module should create the s3 bucket in whichever region I specify. 

## Steps to Reproduce
1. Create a `main.tf` file similar to https://gist.github.com/discentem/427f3dddf11863c2528b5859d1fec1d3 and run `terraform init`, `terraform plan`, and `terraform apply -auto-approve` (as per https://github.com/cloudposse/terraform-aws-tfstate-backend#usage) 

## Screenshots
If applicable, add screenshots or logs to help explain your problem.

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: Windows 10
 - Terraform version: v0.14.7

**Comment by discentem** [comment]
This seems similar to https://github.com/cloudposse/terraform-aws-tfstate-backend/issues/27 but I can't figure out what I'm missing to make this work.

**Comment by discentem** [comment]
I figured out the solution. s3 bucket names need to be globally unique and I guess someone in us-central-1 was already using this module's default bucket name. I provided an override for the name: 

https://gist.github.com/discentem/aabfe0caaea754d55f24a53372a99df7#file-stuff-tf-L40

I'm going to leave this open though, as I'm curious if the module could handle generating partially randomized names by default?
